src/roles/_shaman_helper.py

```diff
@@ -0,0 +1,423 @@
+import itertools
+import random
+import re
+from collections import deque
+
+from src import channels, users, debuglog, errlog, plog
+from src.functions import get_players, get_all_players, get_main_role, get_reveal_role, get_target
+from src.decorators import command, event_listener
+from src.containers import UserList, UserSet, UserDict, DefaultUserDict
+from src.messages import messages
+from src.events import Event
+
+DEATH = UserDict()          # type: Dict[users.User, users.User]
+PROTECTION = UserList()     # type: List[users.User]
+REVEALING = UserSet()       # type: Set[users.User]
+NARCOLEPSY = UserSet()      # type: Set[users.User]
+SILENCE = UserSet()         # type: Set[users.User]
+DESPERATION = UserSet()     # type: Set[users.User]
+IMPATIENCE = UserList()     # type: List[users.User]
+PACIFISM = UserList()       # type: List[users.User]
+INFLUENCE = UserSet()       # type: Set[users.User]
+EXCHANGE = UserSet()        # type: Set[users.User]
+LYCANTHROPY = UserSet()     # type: Set[users.User]
+LUCK = UserSet()            # type: Set[users.User]
+PESTILENCE = UserSet()      # type: Set[users.User]
+RETRIBUTION = UserSet()     # type: Set[users.User]
+MISDIRECTION = UserSet()    # type: Set[users.User]
+DECEIT = UserSet()          # type: Set[users.User]
+
+# holding vars that don't persist long enough to need special attention in
+# reset/exchange/nickchange
+havetotem = []              # type: List[users.User]
+brokentotem = set()         # type: Set[users.User]
+
+# To add new totem types in your custom roles/whatever.py file:
+# 1. Add a key to var.TOTEM_CHANCES with the totem name
+# 2. Add a message totemname_totem to your custom messages.json describing
+#    the totem (this is displayed at night if !simple is off)
+# 3. Add events as necessary to implement the totem's functionality
+#
+# XXX The paragraph below is obsolete, needs to be updated
+#
+# To add new shaman roles in your custom roles/whatever.py file:
+# 1. Expand var.TOTEM_ORDER and upate var.TOTEM_CHANCES to account for the new width
+# 2. Add the role to var.ROLE_GUIDE
+# 3. Add the role to whatever other holding vars are necessary based on what it does
+# 4. Implement custom events if the role does anything else beyond giving totems.
+#
+# Modifying this file to add new totems or new shaman roles is generally never required
+
+def _get_target(var, wrapper, message, lastgiven):
```

Comment:
Missing docstring (as is _give_totem). Name shouldn't start with _ as this is meant to be imported and used by shaman roles.